app/models/lti_deployment.rb

```diff
@@ -2,4 +2,29 @@ class LtiDeployment < ApplicationRecord
   belongs_to :course, optional: true
   belongs_to :lti_client
   validates :external_deployment_id, uniqueness: { scope: :lti_client }
+
+  def get_students
+    auth_data = lti_client.get_oauth_token(['https://purl.imsglobal.org/spec/lti-nrps/scope/contextmembership.readonly'])
+    names_service = LtiService.find_by!(lti_deployment: self, service_type: 'namesroles')
+    membership_uri = URI(names_service.url)
+    membership_uri.query = URI.encode_www_form(role: 'http://purl.imsglobal.org/vocab/lis/v2/membership#Learner')
+    req = Net::HTTP::Get.new(membership_uri)
+    req['Authorization'] = "#{auth_data['token_type']} #{auth_data['access_token']}"
+    http = Net::HTTP.new(membership_uri.host, membership_uri.port)
+
+    res = http.request(req)
+    member_info = JSON.parse(res.body)
+    member_info['members'].each do |member|
+      if member['status'] == 'Inactive'
+        next
+      end
+      user = EndUser.find_by(user_name: member['lis_person_sourcedid'])
+      if user.nil?
+        user = EndUser.create!(user_name: member['lis_person_sourcedid'], first_name: member['given_name'],
```

Comment:
Oh I forgot to mention this earlier, but it would be good to store the student numbers and email addresses as well, especially if we're moving to using Canvas as the source for class lists. 

(The emails should be pretty clear but the student numbers may not be obvious in the Canvas fields and U of T might do something idiosyncratic. If you can't figure it out, please ask the ACT folks.)

Reply:
Email was straightforward. I've asked the ACT folks about student number and am waiting to hear back